### PR TITLE
Fixed Mixed Reality Toolkit service registration & unit tests

### PR DESCRIPTION
**Pull request by StephenHodgson**
Overview
---
Fixed all unit tests and made sure we're registering the correct interfaces by adding a debug error check in MixedRealityToolkit registration 

Changes
---
- Fixes: #2972
- Fixes: #2979

### CODE REVIEW

File: Assets/MixedRealityToolkit/_Core/Services/MixedRealityToolkit.cs [hunk 1]
```diff
@@ -177,7 +177,7 @@ private void Initialize()
                 Utilities.Editor.InputMappingAxisUtility.CheckUnityInputManagerMappings(Definitions.Devices.ControllerMappingLibrary.UnityInputManagerAxes);
```
Comment (SimonDarksideJ) [inline Assets/MixedRealityToolkit/_Core/Services/MixedRealityToolkit.cs]: Can you revert these changes to the MixedRealityToolkit class, as this does not resolve the issues in #2979 
PR #2981 pushed to resolve the underlying issue.

Comment (SimonDarksideJ) [reply Assets/MixedRealityToolkit/_Core/Services/MixedRealityToolkit.cs]: If needed, I can include the "IsAssignableFrom" change, if it's felt this actually helps.

Comment (StephenHodgson) [reply Assets/MixedRealityToolkit/_Core/Services/MixedRealityToolkit.cs]: Will do. And yes, I'd like us to use is assignable from.

**Comment by StephenHodgson** [comment]
Going to close this until service serialization is fixed
